econml/tests/test_driv.py

```diff
@@ -201,34 +209,137 @@ def dgp(n, p, true_fn):
             discrete_treatment=True,
             flexible_model_effect=StatsModelsLinearRegression(fit_intercept=False)
         )]
-
-        # no heterogeneity
-        n = 1000
-        p = 10
-        true_ate = 10
-
-        def true_fn(X):
-            return true_ate
-        y, T, Z, X = dgp(n, p, true_fn)
         for est in ests_list:
             with self.subTest(est=est):
+                # no heterogeneity
```

Comment:
Minor question but is there a benefit to moving this inside the for loop?

Reply:
The test passes :-).  The default is for fit_cov_directly to be True, which means that the previous random seed doesn't generate identical results to what they were before, which lead to a marginal failure on this test.

Logically, I think this makes more sense anyway: it's weird to have different loops creating two sets of identical subtests that test different things; if you run the tests locally via `unittest` you'll see one result per subtest but there won't be any way to tell which was which.